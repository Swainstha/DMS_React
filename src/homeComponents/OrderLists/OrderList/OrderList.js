import React, { useState } from 'react';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';

import CardMenu from '../../../components/CardMenu/CardMenu'
import { Redirect } from 'react-router';

const OrderTableCell = (props) => {

  const [anchorE1, setAnchorE1] = useState(undefined);

  const [menuState, setMenuState] = useState(false);
  const [link, setLink] = useState(null);
  const onOptionMenuSelect = event => {
    setAnchorE1(event.currentTarget);
    setMenuState(true);
  };
  const handleRequestClose = (event, orderId) => {
    console.log(event, orderId);
    setMenuState(false);
    switch (event) {
      case "Update Data":
        setLink(<Redirect to='/app/ims/orders/details'/>);
        break;
      default:
        break;
    }
  };


  const { id, orderId, name, image, orderDate, deliveryDate, from, status } = props.data;
  const statusStyle = status.includes("Completed") ? "text-white bg-success" : status.includes("In Progress") ? "bg-amber" : status.includes("Delayed") ? "text-white bg-danger" : "text-white bg-grey";
  return (

    <tr
      tabIndex={-1}
      key={id}
    >
      <div data-test="link-component">{link}</div>
      <td data-test="orderId-component">{orderId}</td>
      <td>
        <div className="user-profile d-flex flex-row align-items-center">
          <Avatar
            alt={name}
            src={image}
            className="user-avatar"
            data-test="image-component"
          />
          <div className="user-detail">
            <h5 className="user-name" data-test="username-component">{name}</h5>
          </div>
        </div>
      </td>
      <td data-test= "orderdate-component">{orderDate}</td>
      <td data-test= "deliverydate-component">{deliveryDate}</td>
      <td data-test= "from-component">{from}</td>
      <td className="status-cell text-right">
        <div className={` badge text-uppercase ${statusStyle}` } data-test= "status-component">{status}</div>
      </td>
      <td className="text-right">
        <IconButton onClick={onOptionMenuSelect} data-test= "iconbutton-component">
          <i className="zmdi zmdi-more-vert" /></IconButton>
        <CardMenu menuState={menuState} anchorEl={anchorE1} data-test= "cardmenu-component"
          handleRequestClose={(event) => handleRequestClose(event, orderId)} />
      </td>
    </tr>
  );
};

export default OrderTableCell;
