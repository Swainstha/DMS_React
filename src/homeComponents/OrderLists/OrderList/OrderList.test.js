import React from 'react';
import OrderList from './OrderList';
import {
    configure,
    shallow
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const dummy_data= {
    data:{
    id:1, 
    orderId:"33432", 
    name:"Paracetamol", 
    image:"image", 
    orderDate:"5 Aug, 2020", 
    deliveryDate:"7 Aug, 2020", 
    from:"Dharan Hospital", 
    status:"pending"
}
}

const setup = (props = {}, state = null) => {
    return (shallow( < OrderList {...props}  />)
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}



describe('OrderList />', () => {
  
  afterEach(() => {
    jest.clearAllMocks();
  });


it("OrderList snapshot", () => {
    const wrapper = setup(dummy_data)

        // let wrapper = setup(dummy_data);
        expect(wrapper).toMatchSnapshot();

  
})

it("Link Component", () => {
    const wrapper = setup(dummy_data)

   const linkcomponent = findByTestAttr(wrapper, "link-component")
    expect(linkcomponent.text()).toEqual("");
})
it("Components test", () => {
    const wrapper = setup(dummy_data)

    const orderIdcomponent = findByTestAttr(wrapper, "orderId-component")
     expect(orderIdcomponent.text()).toEqual("33432");
     const imageComponent = findByTestAttr(wrapper, "image-component")
     expect(imageComponent.prop("src")).toEqual("image");
     expect(imageComponent.prop("alt")).toEqual("Paracetamol");
     const usernameComponent = findByTestAttr(wrapper, "username-component")
     expect(usernameComponent.text()).toEqual("Paracetamol");
     const orderdatecomponent = findByTestAttr(wrapper, "orderdate-component")
     expect(orderdatecomponent.text()).toEqual("5 Aug, 2020");
     const deliveryDate = findByTestAttr(wrapper, "deliverydate-component")
     expect(deliveryDate.text()).toEqual("7 Aug, 2020");
     const fromComponent = findByTestAttr(wrapper, "from-component")
     expect(fromComponent.text()).toEqual("Dharan Hospital");
 })

 it("IconButton test", () => {
    const wrapper = setup(dummy_data)

    const iconButton = findByTestAttr(wrapper, "iconbutton-component")
    const cardMenu = findByTestAttr(wrapper, "cardmenu-component")
    expect(cardMenu.prop("menuState")).toEqual(false)
    expect(cardMenu.prop("anchorEl")).toEqual(undefined)
    iconButton.props().onClick({currentTarget:"current"})
    wrapper.update()
    const cardMenu2 = findByTestAttr(wrapper, "cardmenu-component")
    expect(cardMenu2.prop("menuState")).toEqual(true)
    expect(cardMenu2.prop("anchorEl")).toEqual("current")
 })

 it("CardMenu test", () => {
     const wrapper = setup(dummy_data)
    const cardMenu = findByTestAttr(wrapper, "cardmenu-component")
    expect(cardMenu.prop("menuState")).toEqual(false)
    expect(cardMenu.prop("anchorEl")).toEqual(undefined)
    const linkcomponent = findByTestAttr(wrapper, "link-component")
    expect(linkcomponent.text()).toEqual("");
    cardMenu.props().handleRequestClose("Update Data")
    wrapper.update()
    const linkcomponent2 = findByTestAttr(wrapper, "link-component")
    expect(linkcomponent2.text()).toEqual("<Redirect />");
    const cardMenu2 = findByTestAttr(wrapper, "cardmenu-component")
    expect(cardMenu2.prop("menuState")).toEqual(false)
    
 })

})
