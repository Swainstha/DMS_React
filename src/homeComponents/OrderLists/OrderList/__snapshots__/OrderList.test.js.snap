// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList /> OrderList snapshot 1`] = `ShallowWrapper {}`;
